presto-main/src/main/java/com/facebook/presto/server/remotetask/ContinuousTaskStatusFetcher.java

```diff
@@ -230,7 +229,8 @@ void updateTaskStatus(TaskStatus newValue)
         AtomicBoolean taskMismatch = new AtomicBoolean();
         taskStatus.setIf(newValue, oldValue -> {
             // did the task instance id change
-            if (!isNullOrEmpty(oldValue.getTaskInstanceId()) && !oldValue.getTaskInstanceId().equals(newValue.getTaskInstanceId())) {
+            boolean isOldValueEmpty = oldValue.getTaskInstanceIdLsb() == 0 && oldValue.getTaskInstanceIdMsb() == 0;
```

Comment:
We can move this logic to the new TaskInstanceId.

Reply:
done